src/codegen/ast_interpreter.cpp

```diff
@@ -1144,9 +1163,17 @@ Box* astInterpretFrom(CompiledFunction* cf, AST_expr* after_expr, AST_stmt* encl
 
     for (const auto& p : locals->d) {
         assert(p.first->cls == str_cls);
-        auto name = static_cast<BoxedString*>(p.first)->s;
-        InternedString interned = cf->clfunc->source->getInternedStrings().get(name);
-        interpreter.addSymbol(interned, p.second, false);
+        std::string name = static_cast<BoxedString*>(p.first)->s;
+        if (name == PASSED_GENERATOR_NAME) {
+            interpreter.setGenerator(p.second);
+        } else if (name == PASSED_CLOSURE_NAME) {
+            interpreter.setPassedClosure(p.second);
+        } else if (name == CREATED_CLOSURE_NAME) {
+            interpreter.setCreatedClosure(p.second);
```

Comment:
Hmm it worries me that we have two places that we can store these names, either in the symbol table or the special slots for them.  This is made worse by the fact that irgen just stores the names as normal names in the symbol table and not in special slots -- if we want to keep slots for perf reasons, maybe we could add an assert somewhere that we're not trying to store/load these special names from the symbol table?